tests/Doctrine/Tests/ORM/Functional/Ticket/Issue5989Test.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\Tests\Models\Issue5989\Issue5989Employee;
+use Doctrine\Tests\Models\Issue5989\Issue5989Manager;
+use Doctrine\Tests\Models\Issue5989\Issue5989Person;
+
+/**
+ * @group issue-5989
+ */
+class Issue5989Test extends \Doctrine\Tests\OrmFunctionalTestCase
+{
+    public function setUp()
+    {
+        $this->useModelSet('issue5989');
+        parent::setUp();
+    }
+
+    public function testSimpleArrayTypeHydratedCorrectlyInJoinedInheritance()
```

Comment:
Besides the functional test, is there a unit test that we can write for the `SimpleObjectHydrator`?

Please don't attempt to do that if it's too complex, just please check it and write one, if it is trivial.